File: src/main/java/com/nerdery/umbrella/data/model/TempUnit.java
package com.nerdery.umbrella.data.model;

import com.google.gson.annotations.SerializedName;

/**
 * Temperature unit to be used in requests for {@link com.nerdery.umbrella.data.api.WeatherService}
 */
public enum TempUnit {
    @SerializedName("si")
    CELSIUS("si"),

    @SerializedName("us")
    FAHRENHEIT("us");

    private String value;

    TempUnit(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: src/main/java/com/nerdery/umbrella/data/constants/TempUnit.kt
package com.nerdery.umbrella.data.constants

import android.content.Context
import android.support.annotation.StringRes
import com.google.gson.annotations.SerializedName
import com.nerdery.umbrella.R

/**
 * Temperature unit to be used in requests for [com.nerdery.umbrella.data.api.WeatherApi]
 */
enum class TempUnit private constructor(
  private val value: String,
  @param:StringRes @field:StringRes private var stringRes: Int
) {
  @SerializedName("si")
  CELSIUS("si", R.string.celsius),

  @SerializedName("us")
  FAHRENHEIT("us", R.string.fahrenheit);

  override fun toString(): String {
    return value
  }

  fun toString(context: Context): String {
    return context.getString(stringRes)
  }
}
